cmd/agent/daemon/state/netflow_pipeline.go

```diff
@@ -58,14 +53,32 @@ func (c *Controller) runNetflowPipeline(ctx context.Context) error {
 	}
 	c.log.Infof("fetched cluster info, pod_cidr=%s, cluster_cidr=%s", c.clusterInfo.podCidr, c.clusterInfo.serviceCidr)
 
+	groupFlows := c.cfg.NetflowGrouping != 0
+
 	errg, ctx := errgroup.WithContext(ctx)
 	errg.Go(func() error {
 		for {
 			select {
 			case <-ctx.Done():
 				return ctx.Err()
 			case e := <-c.tracer.NetflowEvents():
-				c.upsertNetflow(e)
+				if groupFlows {
+					c.upsertNetflow(e)
+				} else {
+					args := e.Args.(types.NetFlowBaseArgs)
```

Comment:
We should probably check if the cast was successful, to not randomly panic.

Reply:
I thought about it but if cast is successful we have a bug in the code. Since it's the only possible valid cast for netflow event type IMO this is safe and panic is fine.